packages/cli/src/commands/setup/auth/auth.js

```diff
@@ -159,10 +246,63 @@ function getRedirectMessage(provider) {
   )}`
 }
 
+// It's safer to run with --ignore-scripts because it won't automatically run
+// post-install scripts. But some packages have legitimate uses of it.
+function yarnAddPackage(module, ignoreScripts = true) {
+  execa.commandSync(
+    `yarn add -D ${module} ${ignoreScripts ? '--ignore-scripts' : ''}`,
+    {
+      stdio: 'inherit',
+      cwd: getPaths().base,
+    }
+  )
+}
+
 /**
  * @param {string} module
  */
-async function getAuthHandler(module) {
+async function getAuthSetup(
+  module,
+  { matchRwVersion = true, ignoreScripts = false } = {}
+) {
+  if (matchRwVersion) {
+    return getMatchingAuthSetup(module, ignoreScripts)
+  }
+
+  // If there's an '@' symbol, and it's not the first character, then we assume
+  // a version is specified
+  if (module.lastIndexOf('@') > 0) {
+    return getSpecificAuthSetup(module, ignoreScripts)
+  }
+
+  return getLatestAuthSetup(module, ignoreScripts)
+}
+
+async function getSpecificAuthSetup(module, ignoreScripts) {
+  const [moduleName, version] = module.split('@')
+  if (!isInstalled(moduleName, version)) {
+    yarnAddPackage(module, ignoreScripts)
+  }
+
+  return await import(module)
+}
+
+async function getLatestAuthSetup(module, ignoreScripts) {
+  const { stdout } = execa.commandSync(
+    `yarn npm info ${module} --fields versions --json`
```

Comment:
This is where the command failed when I tried it, because `@redwoodjs/auth-gotrue` isn't a package